monkey/infection_monkey/monkey.py

```diff
@@ -169,6 +169,26 @@ def _get_arguments(args):
 
         return opts
 
+    @staticmethod
+    def _get_otp():
+        # No need for a constant, this is a feature flag that will be removed.
+        if OTP_FLAG not in os.environ:
+            return "PLACEHOLDER_OTP"
+
+        try:
+            otp = os.environ[AGENT_OTP_ENVIRONMENT_VARIABLE]
+        except KeyError:
+            raise Exception(
+                f"Couldn't find {AGENT_OTP_ENVIRONMENT_VARIABLE} environmental variable."
+                f"Without an OTP the agent will fail to authenticate!"
+            )
+
+        # SECURITY: There's no need to leave this floating around in a place as visible as
+        # environment variables for any longer than necessary.
+        del_key(os.environ, AGENT_OTP_ENVIRONMENT_VARIABLE)
```

Comment:
I don't see a point in wrapping `os.environ.pop(AGENT_OTP_ENVIRONMENT_VARIABLE, None)`. I think we shouldn't have `del_key`

Reply:
It's just defensive coding. If something else modifies the environment between these two statements (highly unlikely but ??), not using `del_key()` would cause the agent to crash.